packages/core/src/deployment.ts

```diff
@@ -1,34 +1,45 @@
 import { promisify } from 'util';
 
 import debug from './utils/debug';
-import { EthereumProvider, getTransactionByHash, getCode, isDevelopmentNetwork } from './provider';
+import { EthereumProvider, getTransactionByHash, hasCode, isDevelopmentNetwork } from './provider';
 
 const sleep = promisify(setTimeout);
 
 export interface Deployment {
   address: string;
-  txHash: string;
+  txHash?: string;
 }
 
 export async function resumeOrDeploy<T extends Deployment>(
   provider: EthereumProvider,
   cached: T | undefined,
   deploy: () => Promise<T>,
 ): Promise<T> {
-  // If there is a deployment stored, we look its transaction up. If the
-  // transaction is found, the deployment is reused.
   if (cached !== undefined) {
-    debug('found previous deployment', cached.txHash);
-    const tx = await getTransactionByHash(provider, cached.txHash);
-    if (tx !== null) {
-      debug('resuming previous deployment', cached.txHash);
-      return cached;
-    } else if (!(await isDevelopmentNetwork(provider))) {
-      // If the transaction is not found we throw an error, except if we're in
-      // a development network then we simply silently redeploy.
-      throw new InvalidDeployment(cached);
+    const { txHash, address } = cached;
+    if (txHash === undefined) {
+      // Deployments can have address but no txHash if they were migrated from OpenZeppelin CLI
+      if (await hasCode(provider, address)) {
```

Comment:
I don't think this should be here. If there is no `txHash`, I would just return the deployment without checking anything. `resumeOrDeploy` makes no guarantee about the contents of the address (equivalently, it makes no guarantee about the success of the transaction, only that it has been mined). `waitAndValidateDeployment` will then check the contents of the address to see if it has code.